src/services/user.service.ts

```diff
@@ -21,4 +22,65 @@ export class UserService implements IUserService {
         return await UserModel.insertMany([userRegisterDTO]);
     }
 
+    public async login(userLoginDTO: UserLoginDTO): Promise<UserLoginResponseDTO> {
+        const user = await UserModel.findOne({email: userLoginDTO.email});
+
+        if (!user?._id) {
+            throw new Unauthorized('wrong email or password');
+        }
+
+        if (!await compare(userLoginDTO.password, user?.password || '')) {
```

Comment:
!await compare(userLoginDTO.password, user?.password || '') it's better to use it like constant, in this case it will be much more clearly what does it mean

Reply:
agreed, updated